contracts/FlowIDTableStaking.cdc

```diff
@@ -836,22 +838,54 @@ pub contract FlowIDTableStaking {
         /// based on the tokens that they have staked
         pub fun payRewards(_ rewardsBreakdownArray: [RewardsBreakdown]) {
 
+            let feeRewards: UFix64 = FlowFees.getFeeBalance()
+            var mintedRewards: UFix64 = 0.0
+
+            // Borrow the fee admin and withdraw all the fees that have been collected since the last rewards payment
+            let feeAdmin = FlowIDTableStaking.borrowFeesAdmin()
+            let feeVault <- feeAdmin.withdrawTokensFromFeeVault(amount: feeRewards)
+
             let allNodeIDs = FlowIDTableStaking.getNodeIDs()
 
             let flowTokenMinter = FlowIDTableStaking.account.borrow<&FlowToken.Minter>(from: /storage/flowTokenMinter)
                 ?? panic("Could not borrow minter reference")
 
             for rewardBreakdown in rewardsBreakdownArray {
                 let nodeRecord = FlowIDTableStaking.borrowNodeRecord(rewardBreakdown.nodeID)
-                nodeRecord.tokensRewarded.deposit(from: <-flowTokenMinter.mintTokens(amount: rewardBreakdown.nodeRewards))
-                emit RewardsPaid(nodeID: rewardBreakdown.nodeID, amount: rewardBreakdown.nodeRewards)
+                let nodeReward = rewardBreakdown.nodeRewards
```

Comment:
I wonder if it would be better to sum up the the tokens we need to mint first, then mint them at once and combine them with the fees vault.

It looks like we're minting for every node and delegator, and every mint will emit an event -- we could get rid of a lot of unneeded events. We could also then take the conditional out of these this codepath and have the payout part be a lot simpler.

Reply:
I didn't do it that way because then we'd have to iterate through everything twice, which might be more excessive than emitting a minted event for every staker. What do you think?

I guess we could just check what the total payout is and subtract the fee vault balance from that and mint the result.